packages/contracts/contracts/router/modules/exchanges/SudoswapModule.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+import {BaseExchangeModule} from "./BaseExchangeModule.sol";
+import {BaseModule} from "../BaseModule.sol";
+import {ISudoswapRouter} from "../../../interfaces/ISudoswapRouter.sol";
+
+contract SudoswapModule is BaseExchangeModule {
+
+    // --- Fields ---
+
+    ISudoswapRouter public immutable SUDOSWAP_ROUTER;
+
+    // --- Constructor ---
+
+    constructor(address owner, address router, address sudoswap)
+        BaseModule(owner)
+        BaseExchangeModule(router) {
+
+        SUDOSWAP_ROUTER = ISudoswapRouter(sudoswap);
```

Comment:
Just an idea, but it's probably going to be more gas-efficient to fill directly on the pool contract rather than going through the Sudoswap router. Like Gem does (https://twitter.com/vasa_develop/status/1580044013906055168) - although we would still have the Reservoir router indirection (but better have one indirection rather than two).

Reply:
todo: selling into a pool